test/unit/org/apache/cassandra/db/compaction/CompactionsCQLTest.java

```diff
@@ -425,11 +441,21 @@ private void readAndValidate(boolean asc) throws Throwable
         }
         catch (Throwable t)
         {
-            assertTrue(t instanceof MarshalException);
-            assertTrue(t.getMessage().contains("Key 22"));
+            assertTrue(t instanceof CorruptSSTableException);
             gotException = true;
+            Throwable cause = t;
+            while (cause != null && !(cause instanceof MarshalException))
+                cause = cause.getCause();
+            assertNotNull(cause);
+            MarshalException me = (MarshalException) cause;
+            assertTrue(me.getMessage().contains("Key 22"));
         }
         assertTrue(gotException);
+        for (SSTableReader sstable : getCurrentColumnFamilyStore().getLiveSSTables())
+        {
+            assertTrue(sstable.isMarkedSuspect());
+            sstable.unmarkSuspect();
+        }
```

Comment:
All of these things are copy pasted three different tests. Could be a little more DRY but it's not a big deal.